File: spring/core/AnnotationDrivenDI/src/main/java/demos/spring/notes/v10/PricingEngineMock.java
package demos.spring.notes.v10;

import demos.spring.notes.common.PricingEngine;
import org.springframework.stereotype.Component;

@Component
public class PricingEngineMock implements PricingEngine {
    public double price(String itemNo, int quantity) {
        return quantity * 5.50;
    }
}

File: spring/core/AnnotationDrivenDIWithKotlin/src/main/kotlin/demos/spring/notes/v4/PricingEngineMock.kt
package demos.spring.notes.v4

import demos.spring.notes.common.PricingEngine
import org.springframework.stereotype.Component

@Component
class PricingEngineMock : PricingEngine {
    override fun price(itemNo: String, quantity: Int): Double {
        return quantity * 5.50
    }
}
